Cargo.toml

```diff
@@ -23,6 +23,7 @@ zeroize = { version = "1.1.0", default-features = false }
 
 [dev-dependencies]
 borsh = "0.7"
+syn = "=1.0.57"
```

Comment:
You can remove this line as syn is not used in this patch

Reply:
I didn't notice your comment about `borch` in the PR description. Would it help to fixate the minor version of the `borsh` here?

Reply:
I don't think so, the `syn` dependency is hidden in https://github.com/near/borsh-rs/blob/3dddc6d7af2e66bf4f5947c0187a9d1f7dae6888/borsh-schema-derive-internal/Cargo.toml#L16. The crate `borsh-schema-derive-internal` author would have to set the exact dependency version like I've done in this PR (`syn = "=1.0.57"`) or there would have to be a `Cargo.lock` with the working version. The broken `syn` version is `1.0.58` so another "solution" would be to wait until the next `syn` version is released (and hopefully) fixed and then tell all `cocoon` users to refresh their `Cargo.lock`. It's a dependency hell a bit with a broken semver principle unfortunately.
But if you see another solution or I'm wrong, I'll be happy to update the code.

Reply:
@MartinKavik I got it. Interesting. Seems like it makes sense to report the issue to the `syn` repo as well.

For now, I would ask you to put this into a separate commit, and I would accept it as soon as possible as a separate PR. Also, could you put a short comment, something like
```
syn = "=1.0.57" # Workaround about borsh's dependency hell: syn v1.0.58 is broken
```
Really appreciate your efforts, man

Reply:
`borsh-schema-derive-internal`'s author is fast, he's adapted the code to the latest `syn`. 
PR: https://github.com/fadeevab/cocoon/pull/6

(I'll rebase & force-push the branch in this PR and change the title once that PR with `borsh` is merged. Then I'll mark this discussion as resolved.)